### PR TITLE
Delay pmon non-critical daemon

### PR DESCRIPTION
**Pull request by yuazhe**
#### Why I did it
After this pull request https://github.com/sonic-net/sonic-buildimage/pull/18295 , the pmon has been added to the start list in fast/warm reboot scenarios. However, certain non-critical daemons of pmon could be delayed, resulting in a saving of approximately 1 second in the reboot process. For performance considerations, especially as the current time usage of fast reboot is closer to 30 seconds limitation, this change could ease the pressure.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
add a script as fast/warm reboot monitor and relative supervisord rlues.
once the script exited means the reboot process has ended, other delayed daemon would then initialize.

#### How to verify it
check the fast/warm reboot time usage

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [x] 202311
- [ ]

### CODE REVIEW

File: dockers/docker-platform-monitor/delay.py
Comment (Junchao-Mellanox) [inline dockers/docker-platform-monitor/delay.py:1]: how about adding some comments to this file to describe the purpose of this daemon?

File: dockers/docker-platform-monitor/docker-pmon.supervisord.conf.j2
Comment (Junchao-Mellanox) [inline dockers/docker-platform-monitor/docker-pmon.supervisord.conf.j2:1]: Why not add delay for pcied? Also, we could discuss with MSFT about if we need delay ycabled.

File: dockers/docker-platform-monitor/delay.py
Comment (yuazhe) [reply dockers/docker-platform-monitor/delay.py:1]: added

File: dockers/docker-platform-monitor/docker-pmon.supervisord.conf.j2
Comment (yuazhe) [reply dockers/docker-platform-monitor/docker-pmon.supervisord.conf.j2:1]: added the delay of pcied

**Review by Junchao-Mellanox — APPROVED** [review]

**Pull request closed without merging**